adcc/backends/pyscf.py

```diff
@@ -45,23 +45,92 @@ def electric_dipole(self):
             -1.0 * self.scfres.mol.intor_symmetric('int1e_r', comp=3)
         )
 
-    @cached_property
+    @property
     def magnetic_dipole(self):
         """-0.5 * sum_i r_i x p_i"""
-        # TODO: Gauge origin?
-        with self.scfres.mol.with_common_orig([0.0, 0.0, 0.0]):
-            return list(
-                -0.5 * self.scfres.mol.intor('int1e_cg_irxp', comp=3, hermi=2)
-            )
+        def g_origin_dep_ints_mag_dip(gauge_origin="origin"):
+            gauge_origin = _transform_gauge_origin_to_xyz(self.scfres, gauge_origin)
+            with self.scfres.mol.with_common_orig(gauge_origin):
+                return list(
+                    -0.5 * self.scfres.mol.intor('int1e_cg_irxp', comp=3, hermi=2)
+                )
+        return g_origin_dep_ints_mag_dip
 
     @cached_property
     def electric_dipole_velocity(self):
         """-sum_i p_i"""
-        with self.scfres.mol.with_common_orig([0.0, 0.0, 0.0]):
+        with self.scfres.mol.with_common_orig((0.0, 0.0, 0.0)):
             return list(
                 self.scfres.mol.intor('int1e_ipovlp', comp=3, hermi=2)
             )
 
+    @property
+    def electric_quadrupole(self):
+        """-sum_i r_{i, alpha} r_{i, beta}"""
+        def g_origin_dep_ints_el_quad(gauge_origin):
+            gauge_origin = _transform_gauge_origin_to_xyz(self.scfres, gauge_origin)
+            with self.scfres.mol.with_common_orig(gauge_origin):
+                r_r = -1.0 * self.scfres.mol.intor_symmetric('int1e_rr', comp=9)
+                return list(r_r)
+        return g_origin_dep_ints_el_quad
+
+    @property
+    def electric_quadrupole_traceless(self):
+        """
+        -0.5 * sum_i (r_{i, alpha} r_{i, beta} - delta_{alpha, beta} r^2)
+        """
+        def g_origin_dep_ints_el_quad_traceless(gauge_origin):
+            gauge_origin = _transform_gauge_origin_to_xyz(self.scfres, gauge_origin)
+            with self.scfres.mol.with_common_orig(gauge_origin):
+                r_r = self.scfres.mol.intor_symmetric('int1e_rr', comp=9)
+                r_r = np.reshape(r_r, (3, 3, r_r.shape[1], r_r.shape[1]))
+                r_quadr = self.scfres.mol.intor_symmetric('int1e_r2', comp=1)
+                r_quadr_matrix = np.zeros_like(r_r)
+                for i in range(3):
+                    r_quadr_matrix[i][i] = r_quadr
+                term = 0.5 * (3 * r_r - r_quadr_matrix)
+                term = -1.0 * np.reshape(term, (9, r_quadr.shape[0],
+                                                r_quadr.shape[0]))
+                return list(term)
+        return g_origin_dep_ints_el_quad_traceless
+
+    @property
+    def electric_quadrupole_velocity(self):
+        """
+        -sum_i (r_{i, beta} p_{i, alpha} - delta_{alpha, beta} S
```

Comment:
I don't think the S is necessary here. And what about the imaginary unit in that term? Maybe we should also generally mention that the imaginary part is implemented?